staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go

```diff
@@ -303,3 +305,12 @@ func FromAPIVersionAndKind(apiVersion, kind string) GroupVersionKind {
 	}
 	return GroupVersionKind{Kind: kind}
 }
+
+type APILifecycle struct {
+	IntroducedVersion *version.Version
+	RemovedVersion    *version.Version
+}
+
+func (v APILifecycle) DeepEqual(other APILifecycle) bool {
+	return v.IntroducedVersion.String() == other.IntroducedVersion.String() && v.RemovedVersion.String() == other.RemovedVersion.String()
```

Comment:
Handle absence of optional fields?

Reply:
Version.String() already handles optional fields

Reply:
Ah.

Should we avoid serializing to string to do a compare?  I'm impressed that this works, but I'm surprised by how it ends up being implemented.

Reply:
Added a `EqualTo` method to compare.